lib/net.c

```diff
@@ -54,6 +52,23 @@ uint8_t default_rss_key[GATEKEEPER_RSS_KEY_LEN] = {
 	0x6a, 0x42, 0xb7, 0x3b, 0xbe, 0xac, 0x01, 0xfa,
 };
 
+
+/* Function to randomize the default RSS key */
+void randomize_rss_key(){
+	srand(time(NULL));
+
+	/* key_initiator gets a random value each time which is xor'd with other values */
+	uint8_t key_intiator = rand();
+	default_rss_key[0] = default_rss_key[0]^key_intiator;
+
+	for(int i=1; i<GATEKEEPER_RSS_KEY_LEN; i++)
+	{
+		default_rss_key[i] = default_rss_key[i]^default_rss_key[i-1];
```

Comment:
This randomization isn't safe since it only produces at most 256 distinct RSS keys. It's simpler and safer to just randomize all bits of the key.

Reply:
The default length of the RSS key is 40 bytes. Here each byte is XOR'd bitwise with the previous byte except for the first byte which is XOR'd with the `key_initiator` which is generated randomly using `rand()` function. So for each byte, there are 256 permutations(2 raised to 8) and for the entire key, the total permutations will be 2 raised to the power 320(40 bytes x 8).

Please correct me if I am wrong.

Reply:
As you point out, only the first byte is random. All the other bytes are mechanically generated from completely known information; in other words, once the first byte is known, all the other bytes are fixed. Therefore, there are at most 256 RSS keys since there are 256 unique values for the first byte. I use "at most" because one or more of these keys can be a duplicate.